src/clients/python/src/tigerbeetle_python/lib.py

```diff
@@ -0,0 +1,84 @@
+import ctypes
+import dataclasses
+import platform
+from pathlib import Path
+
+
+class NativeError(Exception):
+    pass
+
+class IntegerOverflowError(ValueError):
+    pass
+
+
+def _load_tbclient():
+    prefix = ""
+    arch = ""
+    system = ""
+    linux_libc = ""
+    suffix = ""
+
+    platform_machine = platform.machine().lower()
+
+    if platform_machine == "x86_64" or platform_machine == "amd64":
+        arch = "x86_64"
+    elif platform_machine == "aarch64" or platform_machine == "arm64":
+        arch = "aarch64"
+    else:
+        raise NativeError("Unsupported machine: " + platform.machine())
+
+    if platform.system() == "Linux":
+        prefix = "lib"
+        system = "linux"
+        suffix = ".so"
+        libc = platform.libc_ver()[0]
+        if libc == "glibc":
+            linux_libc = "-gnu.2.27"
+        elif libc == "musl":
+            linux_libc = "-musl"
+        else:
+            raise NativeError("Unsupported libc: " + libc)
+    elif platform.system() == "Darwin":
+        prefix = "lib"
+        system = "macos"
+        suffix = ".dylib"
+    elif platform.system() == "Windows":
+        system = "windows"
+        suffix = ".dll"
+    else:
+        raise NativeError("Unsupported system: " + platform.system())
+
+    source_path = Path(__file__)
+    source_dir = source_path.parent
+    library_path = source_dir / "lib" / f"{arch}-{system}{linux_libc}" / f"{prefix}tb_client{suffix}"
+    return ctypes.CDLL(str(library_path))
+
+
+def validate_uint(*, bits: int, name: str, number: int):
+    if number > 2**bits - 1:
+        raise IntegerOverflowError(f"{name}=={number} is too large to fit in {bits} bits")
+    if number < 0:
+        raise IntegerOverflowError(f"{name}=={number} cannot be negative")
+
+
+class c_uint128(ctypes.Structure): # noqa: N801
+    _fields_ = [("low", ctypes.c_uint64), ("high", ctypes.c_uint64)] # noqa: RUF012
+
+    @classmethod
+    def from_param(cls, obj):
+        if obj > (2**128 - 1):
+            raise IntegerOverflowError("Integer Overflow: " + str(obj))
+
+        return cls(high=obj >> 64, low=obj & 0xffffffffffffffff)
+
+    def to_python(self):
+        return self.high << 64 | self.low
```

Comment:
> 🤔 does << work?

It does - ctypes structs play pretty fast and loose with integer types... You define the type as a `c_uint64`, but that's only for information when passed across the FFI boundary. Otherwise, it's always a Python int - both when being called from Python -> C and when being called from C -> Python) _unless_ you explicitly instantiate the type like you did there.

That's why the bindings don't convert from ctypes to Python ints too.